Editors/Audio/Presentation/AudioEditor/AudioProjectData.cs

```diff
@@ -0,0 +1,141 @@
+﻿using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Text;
+using Editors.Audio.Storage;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Shared.Core.PackFiles;
+using Shared.Core.PackFiles.Models;
+
+namespace Editors.Audio.Presentation.AudioEditor
+{
+    public class AudioProjectData
+    {
+        public ProjectSettings Settings { get; set; } = new ProjectSettings();
+        public List<DialogueEventItems> DialogueEvents { get; set; } = new List<DialogueEventItems>();
+
+        public class ProjectSettings
+        {
+            public string BnkName { get; set; }
+            public string Language { get; set; }
+        }
+
+        public class DialogueEventItems
+        {
+            public string DialogueEvent { get; set; }
+            public List<DecisionTreeItems> DecisionTree { get; set; } = new List<DecisionTreeItems>();
+        }
+
+        public class DecisionTreeItems
+        {
+            public string StatePath { get; set; }
+            public List<string> AudioFiles { get; set; } = new List<string>();
+        }
+
+        public static void AddAudioProjectToPackFile(PackFileService packFileService, Dictionary<string, List<Dictionary<string, object>>> eventsData)
+        {
+            var audioProjectJson = ConvertEventDataToAudioProject(eventsData);
+            Debug.WriteLine($"audioProjectJson: {audioProjectJson}");
+
+            var pack = packFileService.GetEditablePack();
+            var byteArray = Encoding.ASCII.GetBytes(audioProjectJson);
+            packFileService.AddFileToPack(pack, "AudioProjects", new PackFile($"audio_project.json", new MemorySource(byteArray)));
+        }
+
+        public static string ConvertEventDataToAudioProject(Dictionary<string, List<Dictionary<string, object>>> eventsData)
+        {
+            var audioProject = new JObject();
+
+            var settings = new JObject
+            {
+                ["BnkName"] = "battle_vo_conversational__ovn_vo_actor_Albion_Dural_Durak",
+                ["Language"] = "english(uk)"
+            };
+
+            audioProject["Settings"] = settings;
+
+            var dialogueEvents = new JArray();
```

Comment:
Do not use the Jarray and other related json based data sources directly. They generally only make problems in life

Reply:
https://learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/how-to

Reply:
Interesting. Will take a look at this.